enhancements/cloud-controller-manager/openstack-cloud-controller-manager.md

```diff
@@ -0,0 +1,173 @@
+---
+title: openstack-cloud-controller-manager
+authors:
+  - "@mfedosin"
+reviewers:
+  - "@crawford"
+  - "@derekwaynecarr"
+  - "@enxebre"
+  - "@eparis"
+  - "@mrunalp"
+  - "@sttts"
+approvers:
+  - "@crawford"
+  - "@derekwaynecarr"
+  - "@enxebre"
+  - "@eparis"
+  - "@mrunalp"
+  - "@sttts"
+creation-date: 2020-08-04
+last-updated: 2020-08-04
+status: implementable
+---
+
+# OpenStack Cloud Controller Manager
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposal describes the migration of OpenStack platform from the deprecated [in-tree cloud provider](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#openstack) to the [Cloud Controller Manager](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-openstack-cloud-controller-manager.md#get-started-with-external-openstack-cloud-controller-manager-in-kubernetes) service that implements `external cloud provider` interface.
+
+## Motivation
+
+Using Cloud Controller Managers (CCMs) is the Kubernetes' [preferred way](https://kubernetes.io/blog/2019/04/17/the-future-of-cloud-providers-in-kubernetes/) to interact with underlying cloud platforms as it provides more flexibility and freedom for developers. It replaces existing in-tree cloud providers, which have been deprecated and will be permanently removed in Kubernetes 1.21. But they are still used in OpenShift and we must start a smooth migration towards CCMs. As a pioneer platform, it is proposed to use OpenStack.
```

Comment:
It's especially important for OpenStack to switch to the external cloud provider because it fixes many issues and limitations with the legacy cloud provider, such as it's reliance on nova-metadata service. For openstack platform, this means support for deploying on provider networks, and deployments at the edge.

Another motivation is to be closer to upstream, benefiting both openshift and k8s.

Reply:
added your text :) thank you!